src/plugins/discover/public/application/main/components/chart/discover_chart.tsx

```diff
@@ -59,6 +59,7 @@ export function DiscoverChart({
   interval?: string;
 }) {
   const isTimeBased = indexPattern.isTimeBased();
+  const isRollupDataView = !!indexPattern.typeMeta?.params?.rollup_index;
```

Comment:
this works, one thing, it could be simplified:

https://github.com/elastic/kibana/blob/cead7b48184698b0d34ae4a08986b09cedeae14c/src/plugins/discover/public/application/main/components/layout/discover_layout.tsx#L125-L127

`isTimeBased` is defined in the parent component, and considers if it's a rollup.
Propagating this variable to `discover_chart.tsx` would be enough to make it functional, no more changes would be needed, `isRollupDataView` would not be necessary